app/plugin_api.go

```diff
@@ -661,11 +661,16 @@ func (api *PluginAPI) GetProfileImage(userID string) ([]byte, *model.AppError) {
 }
 
 func (api *PluginAPI) SetProfileImage(userID string, data []byte) *model.AppError {
-	_, err := api.app.GetUser(userID)
-	if err != nil {
+	if _, err := api.app.GetUser(userID); err != nil {
 		return err
 	}
 
+	if storedData, err := api.app.ReadFile("users/" + userID + "/profile.png"); err == nil {
+		if buf, err := api.app.AdjustImage(bytes.NewReader(data)); err == nil && bytes.Equal(storedData, buf.Bytes()) {
```

Comment:
`SetProfileImageFromFile` already calls `AdjustImage`. Should we move the logic into that? Otherwise this just calls `AdjustImage` twice which looks like a costly operation.

Reply:
We can do that. I was slightly worried of extending the check to the underlying function cause that would affect everything, not only bots initializations.

Reply:
Should be fine IMO. A lot of our store code has similar logic for return-if-same.

Reply:
Well we wouldn't know until we do it and break something anyway :p Will update.